web/components/admin/DirectoryOverview.tsx

```diff
@@ -0,0 +1,162 @@
+import { addDays, addMonths, addYears, format } from "date-fns";
+import { useState } from "react";
+import { useAdminDashDirectoryOverviewQuery } from "../../generated/graphql";
+import { BxsReport } from "../../generated/icons/solid";
+import { useCurrentProfile } from "../../hooks/useCurrentProfile";
+import { useCurrentSpace } from "../../hooks/useCurrentSpace";
+import { Button, Select, Text } from "../atomic";
+import { Responsive } from "../Responsive";
+
+enum Directory_Overview_Date_Range_Enum {
+  PastWeek = "week",
+  PastMonth = "month",
+  PastYear = "year",
+}
+
+const MAP_DATE_RANGE_TO_TEXT = {
+  [Directory_Overview_Date_Range_Enum.PastWeek]: "week",
+  [Directory_Overview_Date_Range_Enum.PastMonth]: "month",
+  [Directory_Overview_Date_Range_Enum.PastYear]: "year",
+};
+
+const MAP_DATE_RANGE_TO_VIEW_DURATION = {
+  [Directory_Overview_Date_Range_Enum.PastWeek]: "Weekly View",
+  [Directory_Overview_Date_Range_Enum.PastMonth]: "Monthly View",
+  [Directory_Overview_Date_Range_Enum.PastYear]: "Yearly View",
+};
+
+const dateToday = new Date();
+const computeAfterDate = (dateRange: Directory_Overview_Date_Range_Enum) => {
+  switch (dateRange) {
+    case Directory_Overview_Date_Range_Enum.PastWeek:
+      return addDays(dateToday, -7);
+    case Directory_Overview_Date_Range_Enum.PastMonth:
+      return addMonths(dateToday, -1);
+    case Directory_Overview_Date_Range_Enum.PastYear:
+      return addYears(dateToday, -1);
+    default:
+      throw new Error("Invalid date range");
+  }
+};
+
+// boxes that display each Directory activity stat
+function DirectoryOverviewInfoBox(props: {
+  amount: number | string;
+  label: string;
+}) {
+  return (
+    <div className="grow flex flex-col justify-start bg-gray-50 p-4 rounded-lg grow">
+      <Text variant="heading3">{props.amount}</Text>
+      <Text variant="body2">{props.label}</Text>
+    </div>
+  );
+}
+
+export function DirectoryOverview() {
+  const { currentSpace } = useCurrentSpace();
+  const { currentProfile } = useCurrentProfile();
+
+  const [DirectoryOverviewDateRange, setDirectoryOverviewDateRange] =
+    useState<Directory_Overview_Date_Range_Enum>(
+      Directory_Overview_Date_Range_Enum.PastWeek
+    );
+
+  // hack
+  const afterDate = computeAfterDate(DirectoryOverviewDateRange);
+
+  const [{ data: adminDashData }, refetchAdminDashData] =
+    useAdminDashDirectoryOverviewQuery({
+      variables: {
+        space_id: currentSpace?.id ?? "",
+        after: afterDate.toISOString(),
+      },
+    });
+
+  const dateRangeDropdownOptions = Object.values(
```

Comment:
nit: move this outside of render function so it doesnt run every render

and since its a const u can do ALL_CAPS capitalization for the var name